src/frontend/qt_sdl/EmuSettingsDialog.cpp

```diff
@@ -252,6 +253,35 @@ void EmuSettingsDialog::on_btnFirmwareBrowse_clicked()
     ui->txtFirmwarePath->setText(file);
 }
 
+void EmuSettingsDialog::on_btnFirmwareEdit_clicked()
+{
+    QDialog dialog(this);
+    QFormLayout form(&dialog);
+
+    QLineEdit* nameEdit = new QLineEdit(&dialog);
+    nameEdit->setText(Config::FirmwareUserName);
+    form.addRow("Username", nameEdit);
+
+    QStringList languages { "Auto", "Japanese", "English", "French", "German", "Italian", "Spanish" };
+
+    QComboBox* languagesBox = new QComboBox(&dialog);
+    languagesBox->addItems(languages);
+    languagesBox->setCurrentIndex(Config::FirmwareLanguage + 1);
+    form.addRow("Language", languagesBox);
```

Comment:
could you add the remaining settings (birthday, message and favourite color) as well?